src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -627,15 +627,85 @@ void ov::npuw::CompiledModel::log_device_dist() const {
 void ov::npuw::CompiledModel::implement_properties() {
     // This function fills the map: {`property name`: `getter for property value`},
     // that can be used later to return requested properties by user.
-    // It does it in 4 steps:
+    // It does it in 3 steps:
     //
-    // 1. Create mappings for all NPUW-specific properties to getters of their
+    // 1. Create mappings for all coped from HETERO plugin properties.
+    // 2. Fill `m_all_supported_props` with properties from 1st step.
+    // 3. Create mappings for all NPUW-specific properties to getters of their
     //    values from config.
-    // 2. Create mappings for all copied from HETERO plugin properties, to
-    //    their copied implementations.
-    // 3. Fill `m_all_supported_props` with all properties, mentioned above.
 
     // 1.
+    m_prop_to_opt = {
+        {ov::supported_properties.name(),
+         {ov::PropertyMutability::RO,
+          [&](const ::intel_npu::Config&) {
+              return m_all_supported_props;
+          }}},
+        {ov::device::properties.name(),
```

Comment:
Do we need this for NPUW? From what I know it was meant for proxy plugins, to provide different set of properties for each device: https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/properties.hpp#L905

Reply:
Thank you a lot, fixed!